src/cf.data.pre

```diff
@@ -1275,9 +1275,80 @@ ENDIF
+	  # identification matching, triggers, and caching (common to both ACLs).
+
 	acl aclname ident_regex [-i] pattern ...
-	  # string match on ident output [slow]
-	  # use REQUIRED to accept any non-null ident.
+	  # regex match against the ident user identifier [slow]
+	  #
+	  # A configured '^-$' pattern (or any equivalent regex) matches:
+	  #   - Transactions associated with a failed user identification attempt.
+	  #   - A successfully identified client-to-Squid connection user with a
+	  #     a user-id equal to a single dash character.
+	  #
+	  # Unlike ident ACL, ident_regex ACL does not treat the REQUIRED pattern
+	  # value specially.
+	  #
+	  # The following paragraphs apply to both ident_regex and ident ACLs.
+	  #
+	  # Under normal conditions, the first of the following events triggers an
+	  # Ident transaction with an Authentication Server (a.k.a. Ident server;
+	  # see RFC 1413) to identify the user behind the client-to-Squid TCP
+	  # connection:
+	  #   - an ident ACL is evaluated
+	  #   - an ident_regex ACL is evaluated
+	  #   - an external ACL with %ui or deprecated %IDENT parameter is evaluated
+	  #   - ident_lookup_access allow rule matches
+	  #
+	  # The results of that Ident transaction (successful or otherwise) are
+	  # stored in Ident cache for the client-to-Squid TCP connection lifetime.
+	  # Concurrent ACL evaluations (associated with the same connection)
+	  # wait for and then reuse cached results. Subsequent ACL evaluations
+	  # reuse cached results immediately, without revalidation.
+	  #
+	  # Successful Authentication Server transactions cache the received
+	  # user-id component of the ident-reply response (see RFC 1413).
+	  #
+	  # Failed Authentication Server transactions cache failure indication.
+	  # Failures include (but are not limited to) cases where Authentication
+	  # Server responds with an invalid (e.g., empty) user-id component.
+	  #
+	  # The following problematic ACL evaluation contexts do not result in an
+	  # ACL match or mismatch. Instead, Squid aborts evaluating rules for the
+	  # configured directive and follows hard-coded directive-specific logic
+	  # (e.g., http_reply_access is denied but miss_access is allowed).
+	  #
+	  #   - ACL evaluation context lacks client-to-Squid connection info. This
+	  #     happens when the ACL is evaluated for transactions that failed to
+	  #     accept a TCP connection or that are not associated with an
+	  #     accepted TCP connection. Squid logs a level-1 ERROR about its
+	  #     inability to "start ident lookup" in this case.
+	  #
+	  #   - Squid has not started (or started but has not finished) obtaining
+	  #     user identity yet and the corresponding configuration directive
+	  #     does not support "slow" ACLs.
+	  #
+	  # In the above problematic cases, Squid does not start a transaction
+	  # with an Authentication Server and does not update Ident cache.
+	  #
+	  # Ident cache is not shared among SMP workers.
 
```

Comment:
@eduard-bagdasaryan, I tried to fix cf.data.pre documentation to match the current branch code (commit 60c8db12). Please see if you can find any remaining discrepancies or other documentation problems.

Please also help me fill out the following table that documents what various ACL configurations do in three special cases (Ident server has erred, Ident server sent an empty user-id component, and Ident server sent a `-` user-id). I entered values for PR code (please double check them!) but left `???` marks for official code.

| squid.conf             | Ident error  | empty user-id | dash user-id |
|---|---|---|---|
|`acl x ident -`         | ???/mismatch | ???/mismatch  | ???/mismatch|
|`acl x ident REQURIED`  | ???/mismatch | ???/mismatch  | ???/mismatch|
|`acl x ident_regex ^-$` | ???/match    | ???/match     | ???/match   |
|`acl x ident_regex ^$`  | ???/mismatch | ???/mismatch  | ???/mismatch|

I table like this should help us summarize functionality changes in this PR (if any). For example, a "match/mismatch" cell value would mean that this PR started to match the corresponding case (where official code mismatched).

Reply:
I ran tests to re-check the values you posted - they are OK. I also checked master - results are the same as PR branch (i.e., the matching result before it asserts). The documentation also looks correct.